Update Eventing Documentation (#14702)
kyma-project/kyma: docs/05-technical-reference/00-custom-resources/evnt-02-eventingbackend.md

---
title: EventingBackend
---

> The EventingBackend custom resource definition (CRD) is used to know the current status of the Eventing backend. To get the up-to-date CRD and show the output in the YAML format, run this command:

Reviewer: ```suggestion
The EventingBackend custom resource definition (CRD) is used to show the current status of the Eventing backend. To get the up-to-date CRD and see the output in the YAML format, run this command:
```

> The EventingBackend custom resource definition (CRD) is used to show the current status of the Eventing backend. To get the up-to-date CRD and see the output in the YAML format, run this command:
> 
> ```shell
> kubectl get crd eventingbackends.eventing.kyma-project.io -o yaml
> ```

Reviewer 2: Currently we only have / support a single instance of the eventing backend custom resource. We have to make this more clear.
```suggestion
The eventing-backend custom resource is used to show the current status of the Eventing Backend. To get the up-to-date CR and see the output in the YAML format, run this command:

```shell
kubectl get crd eventingbackends.eventing.kyma-project.io eventing-backend -n kyma-system -o yaml
```
```

Author: Makes sense!

Author: I get this error though when I run the suggested command:
```
Error from server (NotFound): customresourcedefinitions.apiextensions.k8s.io "eventing-backend" not found
```

I think the command should be:
```
kubectl get eventingbackends.eventing.kyma-project.io eventing-backend -n kyma-system -o yaml
```

## Sample custom resource

> This is a sample custom resource (CR) that the Eventing chart creates by default when Kyma is deployed. It has an empty `spec` section.

Reviewer: ```suggestion
This is a sample custom resource (CR) that the Eventing Controller creates by default when Kyma is deployed. It has an empty `spec` section.
```

```yaml
apiVersion: eventing.kyma-project.io/v1alpha1
kind: EventingBackend
metadata:
  name: eventing-backend
  namespace: kyma-system
spec: {}

> ```

Reviewer 2: As we do not support more than one eventing backend and we also create this as soon as the eventing controller starts i think we do not need this section.

Author: Instead should I paste the output of the previous command:
```
➜  ~ kubectl get eventingbackends.eventing.kyma-project.io eventing-backend -n kyma-system                                                                                                                  (shoot--kymatunas--rp-ir/default)
NAME               BACKEND   EVENTINGREADY   SUBSCRIPTIONCONTROLLERREADY   PUBLISHERPROXYREADY
eventing-backend   NATS      true            True                          True
```

to show the actual usecase of this CR

## Additional information

> When you fetch an existing EventingBackend CR, the system adds the **status** section which describes the status of the Eventing Controller and the Eventing Publisher Proxy. This table lists the fields of the **status** section.

Reviewer: ```suggestion
When you fetch an existing EventingBackend CR, the Eventing Controller adds the **status** section which shows the current status of Kyma Eventing. This table lists the fields of the **status** section.
```

| Field   |  Description |
|:---|:---|
| **status.backendType** | Specifies the backend type used. |
| **status.conditions.code** | Conditions defines the ready status of the Eventing Controller and the Eventing Publisher Proxy. |
| **status.eventingReady** | Current ready status of the Eventing Backend. |

## Related resources and components

These components use this CR:

| Component           | Description                                                                                                  |
| ------------------- | ------------------------------------------------------------------------------------------------------------ |

> | [Eventing Controller](../00-architecture/evnt-01-architecture.md#eventing-controller) | The Eventing Controller uses this CR to display its ready status and the ready status of the [Event Publisher Proxy](../00-architecture/evnt-01-architecture.md#event-publisher-proxy). |

Reviewer: (nit) Instead of Eventing Controller, maybe show the [Subscription](https://kyma-project.io/docs/kyma/main/05-technical-reference/00-custom-resources/evnt-01-subscription/) CR as Related resources and components.

Reviewer: hmmm. agree!

---
title: EventingBackend
---

The `eventingbackends.eventing.kyma-project.io` custom resource definition (CRD) is a detailed description of the kind of data used to manage eventing backends within Kyma. To get the up-to-date CRD and show the output in the YAML format, run this command:

```shell
kubectl get crd eventingbackends.eventing.kyma-project.io -o yaml
```

## Sample custom resource

This is a sample EventingBackend custom resource (CR) that the Eventing Controller creates by default when Kyma is deployed. It has an empty `spec` section.

```yaml
apiVersion: eventing.kyma-project.io/v1alpha1
kind: EventingBackend
metadata:
  name: eventing-backend
  namespace: kyma-system
spec: {}
```

## Additional information

> When you fetch an existing EventingBackend CR, the Eventing Controller adds the **status** section which shows the current status of Kyma Eventing. This table lists the fields of the **status** section.

Reviewer 3: ```suggestion
When you fetch an existing EventingBackend CR, the Eventing Controller adds the **status** section, which shows the current status of Kyma Eventing. The following table lists the fields of the **status** section.
```
